src/test/java/org/olf/folio/order/util/LookupUtilTest.java

```diff
@@ -0,0 +1,116 @@
+package org.olf.folio.order.util;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import org.apache.commons.configuration.CompositeConfiguration;
+import org.apache.commons.configuration.ConfigurationException;
+import org.apache.commons.configuration.PropertiesConfiguration;
+import org.json.JSONObject;
+import org.junit.Before;
+import org.junit.Test;
+import org.olf.folio.order.services.ApiService;
+
+public class LookupUtilTest {
+	
+	CompositeConfiguration config = new CompositeConfiguration();
+	PropertiesConfiguration props = new PropertiesConfiguration();
+    private LookupUtil util;
+    private String token;
+    private String tenant;
+    
+	public LookupUtilTest() {
+		// TODO Auto-generated constructor stub
+	}
+	
+	@Before
+	public void init() {
+		
+		try {
+		    props.load(ClassLoader.getSystemResourceAsStream("application.properties"));
```

Comment:
As is, this returns a `NullPointerException` when attempting to run the app in a container

Relevant snippet from running `docker-compose up --build`:

```log
#12 23.72 -------------------------------------------------------
#12 23.72  T E S T S
#12 23.72 -------------------------------------------------------
#12 24.29 Running org.olf.folio.order.util.LookupUtilTest
#12 24.68 Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.356 sec <<< FAILURE!
#12 24.69 getReferenceValuesTest(org.olf.folio.order.util.LookupUtilTest)  Time elapsed: 0.029 sec  <<< ERROR!
#12 24.69 java.lang.NullPointerException
#12 24.69       at java.base/java.io.Reader.<init>(Reader.java:167)
#12 24.69       at java.base/java.io.InputStreamReader.<init>(InputStreamReader.java:93)
#12 24.69       at org.apache.commons.configuration.AbstractFileConfiguration.load(AbstractFileConfiguration.java:381)
#12 24.69       at org.apache.commons.configuration.AbstractFileConfiguration.load(AbstractFileConfiguration.java:361)
#12 24.69       at org.olf.folio.order.util.LookupUtilTest.init(LookupUtilTest.java:34)
#12 24.69       at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
#12 24.69       at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
#12 24.69       at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
#12 24.69       at java.base/java.lang.reflect.Method.invoke(Method.java:564)
#12 24.69       at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
#12 24.69       at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
#12 24.69       at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
#12 24.69       at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:24)
#12 24.69       at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
#12 24.69       at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
#12 24.69       at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
#12 24.69       at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
#12 24.69       at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
#12 24.69       at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
#12 24.69       at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
#12 24.69       at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
#12 24.69       at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
#12 24.70       at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:252)
#12 24.70       at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:141)
#12 24.70       at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:112)
#12 24.70       at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
#12 24.71       at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:64)
#12 24.71       at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
#12 24.71       at java.base/java.lang.reflect.Method.invoke(Method.java:564)
#12 24.71       at org.apache.maven.surefire.util.ReflectionUtils.invokeMethodWithArray(ReflectionUtils.java:189)
#12 24.71       at org.apache.maven.surefire.booter.ProviderFactory$ProviderProxy.invoke(ProviderFactory.java:165)
#12 24.71       at org.apache.maven.surefire.booter.ProviderFactory.invokeProvider(ProviderFactory.java:85)
#12 24.71       at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:115)
#12 24.71       at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:75)
#12 24.71 
#12 24.73 
#12 24.73 Results :
#12 24.73 
#12 24.73 Tests in error: 
#12 24.73   getReferenceValuesTest(org.olf.folio.order.util.LookupUtilTest)
#12 24.73 
#12 24.73 Tests run: 1, Failures: 0, Errors: 1, Skipped: 0
#12 24.74 
#12 24.75 [INFO] ------------------------------------------------------------------------
#12 24.76 [INFO] BUILD FAILURE
#12 24.76 [INFO] ------------------------------------------------------------------------
#12 24.76 [INFO] Total time:  22.866 s
#12 24.77 [INFO] Finished at: 2021-03-17T18:29:55Z
#12 24.77 [INFO] ------------------------------------------------------------------------
#12 24.77 [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.12.4:test (default-test) on project order-import-poc: There are test failures.
#12 24.77 [ERROR] 
#12 24.77 [ERROR] Please refer to /target/surefire-reports for the individual test results.
#12 24.77 [ERROR] -> [Help 1]
#12 24.77 [ERROR] 
#12 24.77 [ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
#12 24.77 [ERROR] Re-run Maven using the -X switch to enable full debug logging.
#12 24.77 [ERROR] 
#12 24.77 [ERROR] For more information about the errors and possible solutions, please read the following articles:
#12 24.77 [ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
------
executor failed running [/bin/sh -c mvn package]: exit code: 1
ERROR: Service 'order-import-poc' failed to build
```

Reply:
I pulled out code that wrote the list of identifiers to a file (that was for my own debugging).   An application.properties (or .env) files still needs to be created in order to create a container image.  An example file is in the repo without the 'real' username, password, and tenant.

Reply:
No change after you latest commit.

I do have a `.env`, but no `application.properties`.

>  An application.properties (or .env) files still needs to be created in order to create a container image.

Prior to this PR, when building the image in the base branch (`main`), your either/or statement is correct. But with this PR, you have no if/else logic on the truthiness of `USE_SYSTEM_ENV` in `LookupUtilTest`, which introduces a dependency on the `application.properties` as a result.

https://github.com/cul-it/order-import-poc/blob/d16bc8438acefaea7c8ab0597ef32ce81522b127/src/main/java/org/olf/folio/order/listeners/ConfigurationListener.java#L20-L44